src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,364 @@
+@doc raw"""
+    HyperbolicSerreGreenNaghdiEquations1D(bathymetry_type = bathymetry_mild_slope;
+                                          gravity_constant, eta0 = 0.0, λ)
+
+Hyperbolic approximation of the Serre-Green-Naghdi system in one spatial
+dimension. The equations for flat bathymetry are given by
+```math
+\begin{aligned}
+  h_t + (h v)_x &= 0,\\
+  h v_t + \frac{1}{2} g (h^2)_x + \frac{1}{2} h (v^2)_x
+    + \biggl( \frac{\lambda}{3} \eta (1 - \eta / h) \biggr)_x &= 0,\\
+  h w_t + h v w_x &= \lambda (1 - \eta / h),\\
+  \eta_t + \eta_x u &= w.
+\end{aligned}
+\end{aligned}
+```
+The unknown quantities of the hyperbolized Serre-Green-Naghdi equations are the
+total water height ``\eta = h + b`` and the velocity ``v``.
+The gravitational constant is denoted by `g` and the bottom topography
+(bathymetry) ``b = \eta_0 - D``. The water height above the bathymetry
+is therefore given by ``h = \eta - \eta_0 + D``.
+The total water height is therefore given by ``\eta = h + b``.
+
+Two types of variable `bathymetry_type` are supported:
+- [`bathymetry_flat`](@ref): flat bathymetry (typically ``b = 0`` everywhere)
+- [`bathymetry_mild_slope`](@ref): variable bathymetry with mild-slope approximation
+
+For the mild-slope approximation, the Serre-Green-Naghdi equations are
+```math
+\begin{aligned}
+  h_t + (h v)_x &= 0,\\
+  h v_t + \frac{1}{2} g (h^2)_x + \frac{1}{2} h (v^2)_x
+    + \biggl( \frac{\lambda}{3} \eta (1 - \eta / h) \biggr)_x
+    + \biggl( g h + \frac{\lambda}{2} (1 - \eta / h) \biggr) b_x &= 0,\\
+  h w_t + h v w_x &= \lambda (1 - \eta / h),\\
+  \eta_t + \eta_x u + \frac{3}{2} b_x v &= w.
+\end{aligned}
+```
+
+References for the hyperbolized Serre-Green-Naghdi system can be found in
+- Favrie and Gavrilyuk.
+  A rapid numerical method for solving Serre-Green-Naghdi equations
+  describing long free surface gravity waves
+  [DOI: 10.1088/1361-6544/aa712d](https://doi.org/10.1088/1361-6544/aa712d)
+- Busto, Dumbser, Escalante, Favrie, and Gavrilyuk.
+  On High Order ADER Discontinuous Galerkin Schemes for First Order Hyperbolic
+  Reformulations of Nonlinear Dispersive Systems
+  [DOI: 10.1007/s10915-021-01429-8](https://doi.org/10.1007/s10915-021-01429-8)
+
+The semidiscretization implemented here conserves
+- the total water mass (integral of h) as a linear invariant
+- the total modified energy
```

Comment:
```suggestion
- the total water mass (integral of h) as a linear invariant
- the total momentum (integral of h v) as a nonlinear invariant if the bathymetry is constant
- the total modified energy
```
Correct me if I'm wrong.

Reply:
It does not conserve the total momentum - the new non-hydrostatic pressure terms involving $$\eta$$ ($$H$$?) are problematic

Reply:
Yes, you're right. I think I looked at the wrong theorem yesterday.